### PR TITLE
Setup in memory DB in DCD

### PR DESCRIPTION
**Pull request by petterwildhagen**

### CODE REVIEW

**Review by denektenina — APPROVED** [review]
The parts I know about look good for a first version

File: backend/api/api.csproj [hunk 1]
```diff
@@ -2,14 +2,17 @@
 
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
-    <Nullable>enable</Nullable>
+    <Nullable>disable</Nullable>
```
Comment (ReedOnly) [inline backend/api/api.csproj:5]: why disable nullable?

File: backend/api/Services/ProjectService.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+using api.Context;
+using api.Models;
+
+namespace api.Services
+{
+    public class ProjectService
+    {
+        private readonly DCDDbContext _context;
+
+        public ProjectService(DCDDbContext context)
```
Comment (ReedOnly) [inline backend/api/Services/ProjectService.cs:13]: Think it should be DcdDbContext, to not get too much capital letters

File: backend/api/Services/Exceptions.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    public NotFoundInDBException(string message)
+        : base(message)
+    {
+    }
+
+    public NotFoundInDBException(string message, Exception inner)
+        : base(message, inner)
+    {
+    }
+}
```
Comment (ReedOnly) [inline backend/api/Services/Exceptions.cs:18]: not sure if we need our own DB exception class for now

File: backend/api/Program.cs [hunk 4]
```diff
@@ -22,8 +50,8 @@
     app.UseSwaggerUI();
 }
 
-app.UseAuthentication();
-app.UseAuthorization();
+//app.UseAuthentication();
+//app.UseAuthorization();
```
Comment (ReedOnly) [inline backend/api/Program.cs:54]: don't uncomment

File: backend/api/Program.cs [hunk 5]
```diff
@@ -2,12 +2,40 @@
 using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.Identity.Web;
 
+using Microsoft.Data.Sqlite;
+using api.Context;
+using Microsoft.EntityFrameworkCore;
+using api.Services;
```
Comment (ReedOnly) [inline backend/api/Program.cs:8]: we need proper sorting of dependencies

File: backend/api/Controllers/ProjectController.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
+using api.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Identity.Web.Resource;
+using api.Models;
+
+namespace api.Controllers;
+
+//[Authorize]
```
Comment (ReedOnly) [inline backend/api/Controllers/ProjectController.cs:9]: don't uncomment

File: backend/api/Services/Exceptions.cs (hunk 3 above)
Comment (petterwildhagen) [reply backend/api/Services/Exceptions.cs:18]: Deleted. Using Exception instead

File: backend/api/Controllers/ProjectController.cs (hunk 6 above)
Comment (petterwildhagen) [reply backend/api/Controllers/ProjectController.cs:9]: Fixed

File: backend/api/Program.cs (hunk 4 above)
Comment (petterwildhagen) [reply backend/api/Program.cs:54]: Fixed

File: backend/api/Program.cs (hunk 5 above)
Comment (petterwildhagen) [reply backend/api/Program.cs:8]: Fixed

File: backend/api/api.csproj (hunk 1 above)
Comment (petterwildhagen) [reply backend/api/api.csproj:5]: Jeg tok det bort --- da er det tilsvarende BMT. Fikk warnings som Non-nullable property 'ProjectName' must contain a non-null value when exiting constructor. Consider declaring the property as nullable i Models.cs når jeg hadde det og 
Converting null literal or possible null value to non-nullable type. [api]csharp(CS8600) i projectService.cs.
Ser ut som om default er at Nullable = disabled hvis linjen tas ut helt (slik det er i BMT)

File: backend/api/Services/ProjectService.cs (hunk 2 above)
Comment (petterwildhagen) [reply backend/api/Services/ProjectService.cs:13]: Fixed

**Pull request merged**